### PR TITLE
fix: Proper release workflow

### PR DESCRIPTION
**Pull request by ds-pweick**
## Description

This PR changes the workflow `release.yml` in that the workflow now performs what was previously done via a separate workflow `update-docs-for-release.yml` which updated the docs upon release.

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [X] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [X] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -1,72 +1,241 @@
...
       - release
-    uses: ./.github/workflows/publish-swagger-hub.yml
-    with:
-      version: ${{ inputs.new-irs-version }}
-    secrets: inherit
\ No newline at end of file
+    name: "Validate OpenAPI definition"
+    uses: ./.github/workflows/swagger-editor-validate.yml
+    secrets: inherit
+    if: ${{ github.ref_name != 'main' }} # triggered automatically on main
```
Comment (ds-jhartmann) [inline .github/workflows/release.yaml:241]: After the tag is created, the workflows should be run based on the tag, not the main branch. This is especially important for the docker image build step where the image is tagged based on the GitHub ref and the trivy image scan which should check the image build by the image build step

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -194,40 +215,66 @@ https://github.com/google/google-java-format/blob/master/README.md#intellij-jre-
...
+    - IRS release version - type: semantic version string
+    - Helm Chart release version - type: semantic version string
+    - Whether to automatically update the Helm Chart changelog with the change
+      _"- Update IRS version to x.x.x"_. This is a checkbox which is ticked by default.
+5. The workflow makes the automated changes and creates a pull request from them right away.
+   Please review the created pull request within 15 minutes. If you fail to do so, or if the
+   pull request gets closed, the workflow will fail.
+6. Once the pull request has been merged, the workflow continues execution. It pushes a tag (which will be
+   the semantic IRS release version number) and creates the release. Subsequent workflows, such
+   as the IRS build workflow, are triggered automatically.
```
Comment (ds-jhartmann) [inline CONTRIBUTING.md:251]: That's not quite the hint I had in mind. Please just stick to the "new" way of creating a release

**Review by ds-jhartmann — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (ds-pweick) [reply CONTRIBUTING.md]: Hint text now changed to "old way" and "new way"

File: .github/workflows/release.yaml (hunk 1 above)
Comment (ds-pweick) [reply .github/workflows/release.yaml]: Should now be resolved

**Review by ds-jhartmann — APPROVED** [review]
LGTM

**Pull request merged**